### DISCUSSION TITLE
What is different folder structure between Pages Router and App Router

### DISCUSSION THREAD

**Discussion by JuanPatrix** [vercel/next.js#70379] [Help] [answered]
### Summary

I check one from template of Next.Js  https://vercel.com/new/templates/next.js/nextjs-wordpress-headless-cms
is it using Pages Router or App Router?

When I configure page tsx for enable Incremental Static Regeneration (ISR), got this error

### Additional information

```js
Import trace for requested module:
./app/pages/page.tsx
 ⨯ ./app/pages/page.tsx
Error: 
  × "getStaticProps" is not supported in app/. Read more: https://nextjs.org/docs/app/building-your-application/data-fetching
  │ 
  │ 
    ╭─[/Users/juanpatrick/Desktop/Terraform/web-scrapping/ChatBolehWin/next-wp/app/pages/page.tsx:8:1]
  8 │   pages: Page[];
  9 │ }
 10 │ 
 11 │ export const getStaticProps: GetStaticProps<PageProps> = async () => {
    ·              ──────────────
 12 │   const pages = await getAllPages();
 13 │   return {
 14 │     props: {
    ╰────
```

**Comment by Juneezee** [comment]
It is using App Router. The top level folder is `app/`, not `pages/`.

**Comment by icyJoseph** [comment, marked as answer]
You are mixing a few things here. 

`getStaticProps` is not available in App Router.

Rather, you have to get the data for the page as you render the component:

https://github.com/9d8dev/next-wp/blob/75ad41ef59c422429db0bf8b030b829acc97656a/app/pages/page.tsx#L5-L8

Extracted here for visibility:

```tsx
export default async function Page() {
  const pages = await getAllPages();

  return (
    <Section>
      <Container>
        <h1>Pages</h1>

        <h2>All Pages</h2>
        <div className="grid">
          {pages.map((page: any) => (
            <Link key={page.id} href={`pages/${page.slug}`}>
              {page.title.rendered}
            </Link>
          ))}
        </div>
      </Container>
    </Section>
  );
}
```

That's the whole point of App Router.

**_So to answer your question,_** the example there is using App Router, where you can use Server Components, which enable you to fetch the data as part of your render pass.

**Comment by JuanPatrix** [comment]
OK, solved
